daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -143,7 +149,7 @@ class ContractStateMachine[Nid](mode: ContractKeyUniquenessMode) {
           val keys = activeState.keys
           val conflict = keys.get(ck) match {
             case Some(keyMapping) => keyMapping.isDefined
-            case None => globalKeyInputs.get(ck).exists(_.isActive)
+            case None => lookupActiveGlobalKeyInput(ck).exists(_.isDefined)
```

Comment:
@remyhaemmerle-da as discussed you’ll add the testcase from https://gist.github.com/cocreature/092162690ff01efd566cf4740826e069